### ISSUE TITLE
ray[tune] : unpin redis in requirements.txt

### ISSUE THREAD

**Issue by mirekphd** [ray-project/ray#12653]
### What is the problem?
Please unpin `redis` python package in the appropriate requirements file. According to `pipdeptree` it's a tricky one - with an "less than" inequality sign, very easy to escape notice. 

The pinning of `redis` to a previous version causes `pip` to downgrade it immediately after installing `ray[tune]` and our CICD script upon next versions check concludes that `redis` needs an upgrade, leading to unnecessary rebuilds of our three actively monitored python containers during every packages version check, i.e. every hour..:) Currently the quickest but effective workaround was uninstalling `ray` (tested to work in one of the pipelines) and instructing users to use ray from our older docker images.

```
20:35:34  redis :
20:35:34  - version installed:  3.4.1
20:35:34  - latest  available:  3.5.3
20:35:34  - package upgradeable:  True 
[..]
20:48:12  Collecting ray
20:48:12    Downloading ray-1.0.1.post1-cp38-cp38-manylinux1_x86_64.whl (23.1 MB)
[..]
20:48:15  Collecting redis
20:48:15    Downloading redis-3.4.1-py2.py3-none-any.whl (71 kB)
[..]
20:53:11  Successfully installed [..] ray-1.0.1.post1 redis-3.4.1 [..]

 ```

### Reproduction (REQUIRED)
```
$ docker run --rm -it --name ml-gpu-py38-cuda101-cust mirekphd/ml-gpu-py38-cuda101-cust bash
$ pipdeptree -r -p redis
[..]
------------------------------------------------------------------------
redis==3.4.1
  - ray==1.0.1.post1 [requires: redis>=3.3.2,<3.5.0]
```

**Comment by ericl** [comment]
@pcmoritz @simon-mo do we have particular redis requirements?

**Comment by simon-mo** [comment]
It should be bumped to >=3.5.0 already after #11776

**Comment by mirekphd** [comment]
> It should be bumped to >=3.5.0 already after #11776

Yup, it looks like it's just waiting for a new release: https://github.com/ray-project/ray/blob/10572611aa8c9638d97b755c6a7b19afd5e39497/python/setup.py#L157

**Comment by mirekphd** [comment]
Why closing? Not fixed yet IMHO:)
```
00:35:38  redis :
00:35:38  - version installed:  3.4.1
00:35:38  - latest  available:  3.5.3
00:35:38  - package upgradeable:  True
```

**Comment by mirekphd** [comment]
Now (in `ray==1.1.0`) everything looks fine to me!:) We can safely close the issue IMO.

```
20:35:32  ray :
20:35:32  - version installed:  1.1.0
20:35:32  - latest  available:  1.1.0
20:35:32  - package upgradeable:  False 

20:35:32  redis :
20:35:32  - version installed:  3.5.3
20:35:32  - latest  available:  3.5.3
20:35:32  - package upgradeable:  False 
````
